### PR TITLE
Add support for quadratic programs

### PR DESCRIPTION
**Pull request by fuglede**
This adds support for solving quadratic programs on the same footing as (mixed integer) linear programs. HiGHS itself supports these programs out of the box, so the main change necessary here is to ensure that all outputs are parsed as expected.

One complicating factor is that the collection of populated columns is different: Just like outputs for infeasible problems lack some of the columns, for quadratic programs the "Status" column is not populated. We take care of this by following the same pattern as for the other cases, simply filtering the header before parsing the result; to do so, we keep track of whether we are looking at a quadratic program and as a bonus, we add a property called `IsQuadratic` to the output.

A couple of tests are added to ensure that this works as expected. As it turns out, the version of HiGHS added as a submodule produces incorrect results for one of the test cases, by producing non-error output for a problem whose Hessian isn't positive semi-definite. To get around this problem, we simply update the HiGHS submodule to the most recent version and check that the problematic problem is indeed problematic.

Another change since the previously bundled version of HiGHS is that the output now contains the value of the objective as well. While this seems like something that could be useful to include in the JavaScript output object as well, here we just opt to remove the final four lines before parsing, leaving the parsing of the objective value as an exercise for the future.

Some questions here:
* Is `const isQuadratic = model_str.indexOf("[") !== -1;` really the best way to check if this is a QP?
* We say that `isLinear` is `false` when `isQuadratic` is `true`; are those the expected semantics?

This closes gh-13.

### CODE REVIEW

**Comment by lovasoa** [comment]
@jajhall , I think we have had this discussion before. The "pretty" format contains more information than the "computer-readable" one. We'll be able to switch the the coputer-readable one when there will be an easy way to retrieve all the information we need from it.

https://github.com/lovasoa/highs-js/issues/7

**Comment by jajhall** [comment]
> @jajhall , I think we have had this discussion before. The "pretty" format contains more information than the "computer-readable" one. We'll be able to switch the the coputer-readable one when there will be an easy way to retrieve all the information we need from it. See #7

OK, stick with the pretty format. We'll never put out the model bounds as part of the solution as they are model data. Once we get some form of basis information for QPs, the pretty format will change, and I'll try to remember to let you know.

File: src/post.js [hunk 1]
```diff
@@ -86,7 +86,9 @@ Module["solve"] = function (model_str, highs_options) {
     "write and extract solution"
   );
   _Highs_destroy(highs);
-  const output = parseResult(stdout_lines, status);
+  const isQuadratic = model_str.indexOf("[") !== -1;
```
Comment (lovasoa) [inline src/post.js:89]: Is this really correct ? Isn't there a way to detect a quadratic program from the output ?
Is there absolutely no way for a non-quadratic program to contain a `[` ?

**Review by lovasoa — CHANGES_REQUESTED** [review]

**Comment by fuglede** [comment]
> Once we get some form of basis information for QPs, the pretty format will change, and I'll try to remember to let you know.

I suppose it's worth noting that such a change could only possibly break `highs-js` upon a git submodule update, and if it does, the test added here should catch it.

File: src/post.js (hunk 1 above)
Comment (fuglede) [reply src/post.js:89]: Yeah, good question; I pondered the same. [The specification](http://lim.univ-reunion.fr/staff/fred/Enseignement/Optim/doc/CPLEX-LP/CPLEX-LP-file-format.html) says the following:

>  Also, the following characters are not valid in variable names (in order to allow for quadratic objective information): ^, *, [ and ]. 

I believe this makes the check safe (although it'd certainly be more comfortable to be able to pick the information out of the library somehow).

Comment (lovasoa) [reply src/post.js:89]: Yes, we should improve output parsing instead of trying to parse the input.

Comment (fuglede) [reply src/post.js]: I've added a commit changing the logic for determining the headers: Rather than basing them on the type of problem (which was what was hard for the QPs), this instead decides whether or not a particular header should be included by looking at the row immediately below, checking whether there is any data available. The data can be either left or right aligned, so we check both. I added a few more tests to cover some edge cases.

Previously, `isLinear` (and `isQuadratic`) would be used to determine which headers were there, so I can see how one might as well add the information to the output, but with this change, they're not used for much, and I do wonder if there's any benefit from having them in the output (presumably the user would typically know what problem they come with), but chances are there's a use case I'm missing?

File: src/post.js [hunk 2]
```diff
@@ -151,32 +150,50 @@ function lineToObj(headers, line) {
...
+function headersForNonEmptyColumns(headerLine, firstDataLine) {
+  // Headers can correspond to empty columns. The contents of a column can be left or right
+  // aligned, so we determine if a given header should be included by looking at whether
+  // the row immediately below the header has any contents.
+  const headers = [...headerLine.matchAll(/[^\s]+/g)];
+  const headerStarts = headers.map(h => h.index);
+  const headerEnds = headers.map(h => h.index + h[0].length - 1);
+  return headers.map(x => x[0]).filter(
+    (_, i) => firstDataLine[headerStarts[i]] !== ' ' || firstDataLine[headerEnds[i]] !== ' ');
+}
```
Comment (lovasoa) [inline src/post.js:195]: maybe just :

```suggestion
function headersForNonEmptyColumns(headerLine, firstDataLine) {
	// Headers can correspond to empty columns. The contents of a column can be left or right
	// aligned, so we determine if a given header should be included by looking at whether
	// the row immediately below the header has any contents.
	return [...headerLine.matchAll(/[^\s]+/g)].filter(match =>
		firstDataLine[match.index] !== ' ' ||
		firstDataLine[match.index + match[0].length - 1] !== ' '
	).map(match => match[0])
}
```

?

Comment (fuglede) [reply src/post.js:195]: Thanks, much better

File: src/post.js (hunk 1 above)
Comment (fuglede) [reply src/post.js]: And thanks for the swift review!
